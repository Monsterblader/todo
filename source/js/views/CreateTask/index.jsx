import React, { Component } from "react";

export default class CreateTask extends Component {
  state = {
    task: "",
    startDate: "",
    endDate: "",
  };

  handleChange = e => {
    const target = e.target;
    const name = target.name;

    this.setState({[name]: target.value});
  }
  ;

      handleKeyPress = e => {
        if (e.key !== "Enter")
          return;

        const {onSubmitEditing} = this.props;
        const {task, startDate, endDate} = this.state;

        if (!task)
          return; // Don't submit if empty

        onSubmitEditing({task, startDate, endDate});
        this.setState({
          endDate: "",
          index: "",
          parent: "",
          startDate: "",
          task: "",
        });
      }
  ;

      render() {
    const {placeholder} = this.props;
    const {task, startDate, endDate} = this.state;

    return (
        <div>
          <input
            name="task"
            style={styles.input}
          type={"text"}
          value={task}
          placeholder={placeholder}
          onChange={this.handleChange}
          onKeyPress={this.handleKeyPress}
          />
          <input
            name="startDate"
            style={styles.input}
          type={"date"}
          value={startDate}
          placeholder={placeholder}
          onChange={this.handleChange}
          onKeyPress={this.handleKeyPress}
          />
        <input
          name="endDate"
          style={styles.input}
          type={"date"}
          value={endDate}
          placeholder={placeholder}
          onChange={this.handleChange}
          onKeyPress={this.handleKeyPress}
          />
        </div>
        );
  }
}

const styles = {
  input: {
    fontSize: "100%",
    padding: 15,
    borderWidth: 0
  }
};

