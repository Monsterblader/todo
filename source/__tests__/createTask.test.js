import React from 'react';
import { shallow, mount, render } from 'enzyme';
import CreateTask from '../js/views/CreateTask';
import { fromJS } from 'immutable';
import renderer from 'react-test-renderer';

describe('CreateTask', () => {
  const passThroughFn = arg => arg,
    onSubmitEditingSpy = jest.fn(),
    mockValue = fromJS({id: 'a1', index: '0', task: 'task1', startDate: 'startDate1', endDate: 'endDate1'}),
    createTask = shallow(
      <CreateTask
        onSubmitEditing={onSubmitEditingSpy}
        placeholder="test"
      />
    );

  it('should render', () => {
    const createTask = renderer.create(
      <CreateTask
        onSubmitEditing={onSubmitEditingSpy}
        placeholder="test"
      />
    );
    expect(createTask).toMatchSnapshot();
  });

  it('should render with "task," "startDate," and "endDate"', () => {
    expect(createTask.find('input')).toHaveLength(3);
    expect(createTask.find('.enterTask')).toHaveLength(1);
    expect(createTask.find('.enterStartDate')).toHaveLength(1);
    expect(createTask.find('.enterEndDate')).toHaveLength(1);
  });

  it('should handle key presses and field changes', () => {
    const testEvent = {target: {name: 'task', value: 'a'}};
    createTask.instance().handleChange(testEvent);
    expect(createTask.instance().state.task).toEqual('a');
    // const changedTask = createTask.find('.enterTask').simulate('keypress', {key: 128});
    // expect(changedTask.find('.enterTask').html()).toEqual("purple");
  });
});
