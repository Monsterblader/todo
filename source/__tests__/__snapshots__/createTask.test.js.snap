// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateTask should render 1`] = `
<div>
  <input
    className="enterTask"
    name="task"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="test"
    style={
      Object {
        "borderWidth": 0,
        "fontSize": "100%",
        "padding": 15,
      }
    }
    type="text"
    value=""
  />
  <input
    className="enterStartDate"
    name="startDate"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="test"
    style={
      Object {
        "borderWidth": 0,
        "fontSize": "100%",
        "padding": 15,
      }
    }
    type="date"
    value=""
  />
  <input
    className="enterEndDate"
    name="endDate"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="test"
    style={
      Object {
        "borderWidth": 0,
        "fontSize": "100%",
        "padding": 15,
      }
    }
    type="date"
    value=""
  />
</div>
`;
